metacat-main/src/main/java/com/netflix/metacat/main/services/impl/TableServiceImpl.java

```diff
@@ -373,11 +474,24 @@ public Optional<TableDto> get(final QualifiedName name, final GetTableServicePar
         }
 
         if (getTableServiceParameters.isIncludeDefinitionMetadata()) {
-            final Optional<ObjectNode> definitionMetadata =
+            Optional<ObjectNode> definitionMetadata =
                 (getTableServiceParameters.isDisableOnReadMetadataIntercetor())
                     ? userMetadataService.getDefinitionMetadata(name)
                     : userMetadataService.getDefinitionMetadataWithInterceptor(name,
                     GetMetadataInterceptorParameters.builder().hasMetadata(tableInternal).build());
+            // Always get the source of truth for parent child relation from the parentChildRelMetadataService
+            final Set<ParentInfo> parentInfo = parentChildRelMetadataService.getParents(name);
```

Comment:
if getParents() throws an exception, are we comfortable failing the whole getTable call? i believe that is the correct thing to do, but want to call out that this has effects on the majority of tables despite them having no parent-child relationship

Reply:
For now, when getDefinitionMetadata failed, we will also throw an exception so I think it's ok.